crates/lib3h/src/keystore.rs

```diff
@@ -0,0 +1,143 @@
+//! This is a stub of the keystore so we can prove out the encoding transport
+
+use crate::error::{Lib3hError, Lib3hResult};
+use detach::prelude::*;
+use lib3h_ghost_actor::prelude::*;
+use std::any::Any;
+
+pub mod keystore_protocol {
+    #[derive(Debug)]
+    pub enum RequestToChild {
+        Sign { id: String, payload: Vec<u8> },
+    }
+
+    #[derive(Debug)]
+    pub enum RequestToChildResponse {
+        Sign { signature: Vec<u8> },
+    }
+
+    #[derive(Debug)]
+    pub enum RequestToParent {}
+
+    #[derive(Debug)]
+    pub enum RequestToParentResponse {}
+}
+
+pub type KeystoreActor = Box<
+    dyn GhostActor<
+        RequestToParent,
+        RequestToParentResponse,
+        RequestToChild,
+        RequestToChildResponse,
+        Lib3hError,
+    >,
+>;
+pub type KeystoreActorParentEndpoint = GhostEndpoint<
+    RequestToChild,
+    RequestToChildResponse,
+    RequestToParent,
+    RequestToParentResponse,
+    Lib3hError,
+>;
+pub type KeystoreActorParentWrapper<Context> = GhostParentWrapper<
+    Context,
+    RequestToParent,
+    RequestToParentResponse,
+    RequestToChild,
+    RequestToChildResponse,
+    Lib3hError,
+>;
+
+use keystore_protocol::*;
+
+pub struct KeystoreStub {
```

Comment:
Can you use some of the type aliases here instead for clarity / code reduction?

Reply:
totally. done : )